packages/salesforcedx-vscode-apex/src/oas/generationStrategy/generationStrategy.ts

```diff
@@ -15,11 +15,11 @@ import {
 export abstract class GenerationStrategy {
   abstract metadata: ApexClassOASEligibleResponse;
   abstract context: ApexClassOASGatherContextResponse;
-  abstract prompts: string[];
+  abstract prompts: string[] | Map<string, string>; // Map<methodName, prompt>
```

Comment:
Changing the type in the base class forces subclasses to adopt that change as well. If a given strategy requires a different shape, and are only used within the subclass, then I would suggest moving strategy specific need to each subclass and remove these proeprties from the base class.